FlowCrypt/Core/Core.swift

```diff
@@ -60,6 +60,11 @@ final class Core: KeyDecrypter, CoreComposeMessageType {
 
     func decryptKey(armoredPrv: String, passphrase: String) throws -> CoreRes.DecryptKey {
         let r = try call("decryptKey", jsonDict: ["armored": armoredPrv, "passphrases": [passphrase]], data: nil)
+
+        if let error = try? r.json.decodeJson(as: CoreRes.Error.self) {
+            throw CoreError.exception(error.error.message)
+        }
```

Comment:
Can there be a generic one in `_call` like before? All of these errors have a consistent format `{"error": {"message": ".", "stack": "..."}}`. The original `_call` error handling was generic and had logging done too which is important.

The system there should be generic but flexible enough so that you could get finer grained errors out of it for individual methods, but only when needed. Elese you have to add the same error handlers / guards into every method here which is worse.

Reply:
Yes you are right. I was thinking that you need to have strong typing like you have in `MsgBlock.DecryptErr.ErrorType`. Now I see that `CoreRes.Error` has `type` parameter too.